### ISSUE TITLE
Indexing language fails on an empty array

### ISSUE THREAD

**Issue by dainiusjocas** [vespa-engine/vespa#30512]
**Describe the bug**
When an empty list is provided this indexing language script fails. Demo app is [here.](https://github.com/dainiusjocas/vespa-colbert-tricks). When either `null` or non empty array is provided, then feeding goes through. 
Also, this problem is observable during reindexing.
After some errors the reindexing status becomes:
```json
{
  "enabled": true,
  "clusters": {
    "realm": {
      "pending": {},
      "ready": {
        "realm": {
          "readyMillis": 1709814605699,
          "speed": 1.0,
          "cause": "reindexing due to a schema change",
          "startedMillis": 1709813520007,
          "endedMillis": 1709814181631,
          "message": "PROCESSING_FAILURE: ReturnCode(PROCESSING_FAILURE, [from content node 1] Processing failed. Error message: com.yahoo.vespa.indexinglanguage.expressions.VerificationException: For expression 'get_var cnt': Variable 'cnt' not found -- See Vespa log for details.)",
          "state": "pending"
        }
      }
    }
  }
}
```

**To Reproduce**
1. Start Vespa, deploy app
2. Feed document
```shell
echo '{
  "fields": {
    "text": []            
  },
  "put": "id:doc:doc::1"
}' | vespa feed -
```
3. This returns status 500, and see error in the logs
```text
[2024-03-08 07:19:08.668] WARNING container        Container.com.yahoo.docproc.jdisc.DocumentProcessingTask	Processing of put of document id:doc:doc::1. Last call: call to class com.yahoo.docprocs.indexing.IndexingProcessor (id: com.yahoo.docprocs.indexing.IndexingProcessor in indexing) failed at call to class com.yahoo.docprocs.indexing.IndexingProcessor (id: com.yahoo.docprocs.indexing.IndexingProcessor in indexing)\nexception=\ncom.yahoo.vespa.indexinglanguage.expressions.VerificationException: For expression 'get_var chunk_cnt': Variable 'chunk_cnt' not found\n\tat com.yahoo.vespa.indexinglanguage.expressions.GetVarExpression.doVerify(GetVarExpression.java:31)\n\tat com.yahoo.vespa.indexinglanguage.expressions.Expression.verify(Expression.java:176)\n\tat com.yahoo.vespa.indexinglanguage.expressions.VerificationContext.execute(VerificationContext.java:29)\n\tat com.yahoo.vespa.indexinglanguage.expressions.ArithmeticExpression.doVerify(ArithmeticExpression.java:87)\n\tat com.yahoo.vespa.indexinglanguage.expressions.Expression.verify(Expression.java:176)\n\tat com.yahoo.vespa.indexinglanguage.expressions.VerificationContext.execute(VerificationContext.java:29)\n\tat com.yahoo.vespa.indexinglanguage.expressions.StatementExpression.doVerify(StatementExpression.java:72)\n\tat com.yahoo.vespa.indexinglanguage.expressions.Expression.verify(Expression.java:176)\n\tat com.yahoo.vespa.indexinglanguage.expressions.VerificationContext.execute(VerificationContext.java:29)\n\tat com.yahoo.vespa.indexinglanguage.expressions.ScriptExpression.doVerify(ScriptExpression.java:71)\n\tat com.yahoo.vespa.indexinglanguage.expressions.Expression.verify(Expression.java:176)\n\tat com.yahoo.vespa.indexinglanguage.expressions.VerificationContext.execute(VerificationContext.java:29)\n\tat com.yahoo.vespa.indexinglanguage.expressions.IfThenExpression.doVerify(IfThenExpression.java:113)\n\tat com.yahoo.vespa.indexinglanguage.expressions.Expression.verify(Expression.java:176)\n\tat com.yahoo.vespa.indexinglanguage.expressions.VerificationContext.execute(VerificationContext.java:29)\n\tat com.yahoo.vespa.indexinglanguage.expressions.StatementExpression.doVerify(StatementExpression.java:72)\n\tat com.yahoo.vespa.indexinglanguage.expressions.Expression.verify(Expression.java:176)\n\tat com.yahoo.vespa.indexinglanguage.expressions.VerificationContext.execute(VerificationContext.java:29)\n\tat com.yahoo.vespa.indexinglanguage.expressions.ForEachExpression.doVerify(ForEachExpression.java:70)\n\tat com.yahoo.vespa.indexinglanguage.expressions.Expression.verify(Expression.java:176)\n\tat com.yahoo.vespa.indexinglanguage.expressions.VerificationContext.execute(VerificationContext.java:29)\n\tat com.yahoo.vespa.indexinglanguage.expressions.ForEachExpression.doExecute(ForEachExpression.java:50)\n\tat com.yahoo.vespa.indexinglanguage.expressions.Expression.execute(Expression.java:99)\n\tat com.yahoo.vespa.indexinglanguage.expressions.ExecutionContext.execute(ExecutionContext.java:35)\n\tat com.yahoo.vespa.indexinglanguage.expressions.StatementExpression.doExecute(StatementExpression.java:59)\n\tat com.yahoo.vespa.indexinglanguage.expressions.Expression.execute(Expression.java:99)\n\tat com.yahoo.vespa.indexinglanguage.expressions.ExecutionContext.execute(ExecutionContext.java:35)\n\tat com.yahoo.vespa.indexinglanguage.expressions.ScriptExpression.doExecute(ScriptExpression.java:54)\n\tat com.yahoo.vespa.indexinglanguage.expressions.Expression.execute(Expression.java:99)\n\tat com.yahoo.vespa.indexinglanguage.expressions.GuardExpression.doExecute(GuardExpression.java:46)\n\tat com.yahoo.vespa.indexinglanguage.expressions.Expression.execute(Expression.java:99)\n\tat com.yahoo.vespa.indexinglanguage.expressions.ExecutionContext.execute(ExecutionContext.java:35)\n\tat com.yahoo.vespa.indexinglanguage.expressions.StatementExpression.doExecute(StatementExpression.java:59)\n\tat com.yahoo.vespa.indexinglanguage.expressions.Expression.execute(Expression.java:99)\n\tat com.yahoo.vespa.indexinglanguage.expressions.ExecutionContext.execute(ExecutionContext.java:35)\n\tat com.yahoo.vespa.indexinglanguage.expressions.ScriptExpression.doExecute(ScriptExpression.java:54)\n\tat com.yahoo.vespa.indexinglanguage.expressions.Expression.execute(Expression.java:99)\n\tat com.yahoo.vespa.indexinglanguage.expressions.Expression.execute(Expression.java:84)\n\tat com.yahoo.vespa.indexinglanguage.expressions.Expression.execute(Expression.java:56)\n\tat com.yahoo.vespa.indexinglanguage.expressions.Expression.execute(Expression.java:52)\n\tat com.yahoo.docprocs.indexing.DocumentScript.execute(DocumentScript.java:51)\n\tat com.yahoo.docprocs.indexing.IndexingProcessor.processDocument(IndexingProcessor.java:113)\n\tat com.yahoo.docprocs.indexing.IndexingProcessor.process(IndexingProcessor.java:74)\n\tat com.yahoo.docproc.Call.call(Call.java:150)\n\tat com.yahoo.docproc.impl.DocprocExecutor.process(DocprocExecutor.java:117)\n\tat com.yahoo.docproc.jdisc.DocumentProcessingTask.process(DocumentProcessingTask.java:120)\n\tat com.yahoo.docproc.jdisc.DocumentProcessingTask.run(DocumentProcessingTask.java:67)\n\tat java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)\n\tat java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)\n\tat java.base/java.lang.Thread.run(Thread.java:840)\n
```

**Expected behavior**
Feeding should succeed and either no or an empty tensor is created.
What also would be great, would be an easy way to get length of the array in the indexing language.

**Environment (please complete the following information):**
 - OS: macOS, Docker.

**Vespa version**
8.315.19

**Additional context**

Related [issue](https://github.com/vespa-engine/vespa/issues/29690).

**Comment by arnej27959** [comment]
Fixed by https://github.com/vespa-engine/vespa/pull/30680
This should be available in next release (vespa >= 8.322.22).

I also fixed so you should not need the `if (true == true)` inside for_each.

**Comment by dainiusjocas** [comment]
Awesome! Thanks!
